index.js

```diff
@@ -3,6 +3,24 @@
 var clone = require('101/clone')
 var debug = require('debug')('swarmerode')
 var exists = require('101/exists')
+var Promise = require('bluebird')
+var cache = {}
+
+function handleCache (key, cacheFetch, cb) {
+
+  // If CACHE_LENGTH is not set we don't want to cache anything
+  if (!exists(process.env.SWARMERODE_CACHE_LENGTH)) {
+    return cacheFetch(cb)
+  }
+
+  if (!cache[key]) {
+    cache[key] = Promise.fromCallback(cacheFetch)
+    setTimeout(function () {
```

Comment:
`.delay()` only gets called on success. Also we don't want to delay the response. I could do `cacheFetch.catch(noop).delay(TIME).then(`